pkg/lint/discovery.go

```diff
@@ -0,0 +1,232 @@
+package lint
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/giantswarm/config-controller/pkg/generator"
+	"github.com/giantswarm/microerror"
+)
+
+type Discovery struct {
```

Comment:
`Discovery` serves as an abstraction so I don't have to parse stuff a 1000 times.